dtfs-central-api/src/v1/controllers/utilisation-report-service/get-fee-record-details.controller/index.test.ts

```diff
@@ -0,0 +1,89 @@
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import { when } from 'jest-when';
+import { FeeRecordEntityMockBuilder, UtilisationReportEntityMockBuilder } from '@ukef/dtfs2-common';
+import { getFeeRecordDetails, GetFeeRecordDetailsResponseBody } from '.';
+import { mapFeeRecordEntityToDetails } from './helpers';
+import { FeeRecordRepo } from '../../../../repositories/fee-record-repo';
+
+console.error = jest.fn();
+
+jest.mock('./helpers');
+
+describe('get-fee-record-details.controller', () => {
+  describe('getFeeRecordDetails', () => {
+    const reportId = 3;
+    const feeRecordId = 14;
+
+    const getHttpMocks = () =>
+      httpMocks.createMocks({
+        params: { reportId: reportId.toString(), feeRecordId: feeRecordId.toString() },
+      });
+
+    const findSpy = jest.spyOn(FeeRecordRepo, 'findOneByIdAndReportIdWithReport');
+
+    beforeEach(() => {
+      findSpy.mockResolvedValue(null);
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it('responds with a 404 when no fee record with the supplied fee record id and report id can be found', async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      findSpy.mockResolvedValue(FeeRecordEntityMockBuilder.forReport(new UtilisationReportEntityMockBuilder().build()).build());
+      when(findSpy).calledWith(feeRecordId, reportId).mockResolvedValue(null);
+
+      // Act
+      await getFeeRecordDetails(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.NotFound);
+      expect(findSpy).toHaveBeenCalledTimes(1);
+      expect(findSpy).toHaveBeenCalledWith(feeRecordId, reportId);
+    });
+
+    it('responds with a 200', async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      when(findSpy)
+        .calledWith(feeRecordId, reportId)
+        .mockResolvedValue(FeeRecordEntityMockBuilder.forReport(new UtilisationReportEntityMockBuilder().build()).build());
+
+      // Act
+      await getFeeRecordDetails(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+      expect(findSpy).toHaveBeenCalledTimes(1);
+      expect(findSpy).toHaveBeenCalledWith(feeRecordId, reportId);
+    });
+
+    it('responds with the fee record details', async () => {
+      // Arrange
+      const { req, res } = getHttpMocks();
+
+      when(findSpy)
+        .calledWith(feeRecordId, reportId)
+        .mockResolvedValue(FeeRecordEntityMockBuilder.forReport(new UtilisationReportEntityMockBuilder().build()).build());
+
+      const feeRecordDetails = {
+        field1: 'Some value',
+        field2: 'Another value',
+      } as unknown as GetFeeRecordDetailsResponseBody;
```

Comment:
I'm not a fan of this pattern, I think it is actively confusing to have weird data we are typing as something else. Instead, please can we introduce a `aFeeRecordResponseBody` function and call that to just get an arbitrary object of the correct type, it's not really any more effort.